WallsAndHoles/editor.cpp

```diff
@@ -95,17 +97,50 @@ void Editor::makeNewScene()
 
 void Editor::saveMap()
 {
-
+    if(mTileMap==nullptr){
+        QMessageBox messageBox;
+        messageBox.critical(0,"Error","TileMap doesn't exist!");
+        messageBox.setFixedSize(500,200);
+        return;
+    }
+    mTileMap->setDepend(QSharedPointer<TileTemplateSet>(mTileTemplateSet));
+    //mTileMap->updateDepend();
+    if(mTileMap->savePath().isEmpty()){
+        mTileMap->setSavePath(QFileDialog::getSaveFileName(mMainWindow,
+            tr("Save Map"), "/home", tr("Save Files (*.xml)")));
+    }
+    for(SharedTileTemplateSet set: mTileMap->dependencies()){
+        qDebug()<<"save dependencies...";
+        if(set->savePath().isEmpty()){
+            set->setSavePath(QFileDialog::getSaveFileName(mMainWindow,
+                tr("Save Templates"), "/home", tr("Save Files (*.xml)")));
```

Comment:
Again, not to worry about right now, but in the future the saving of TileTemplateSets and TileMaps should be separate processes.